### PR TITLE
chore(supabase): Set up Supabase as backend service and add initial tests (#83)

### LINKED ISSUE #83: chore: setting up supabase as backend service
The objective of this task is to integrate Supabase as the backend service for our React Native mobile application. Supabase is an open-source alternative to Firebase that offers a range of backend features, including a PostgreSQL database, authentication, authorization, storage, and more.

### PR DESCRIPTION
**Pull request by 911samuel**
#### What does this PR do?

This PR sets up Supabase as the backend service for our application and includes initial tests for sign-up and sign-in functionality.

#### Description of Task to be completed?

* Initializes the Supabase client with the provided URL and Anon Key.
* Configures authentication settings for AsyncStorage, enabling auto-refresh and session persistence.
* Adds basic tests to verify sign-up and sign-in functionality.

#### How should this be manually tested?

To manually test the changes:

1. Access the sign-up page (if applicable).
2. Enter a valid email address and password.
3. Confirm the password.
4. Click the "Sign Up" button.
5. Verify that the user is redirected to the user profile page (or expected success behavior).
6. Try to sign up with an invalid email address or password.
7. Ensure appropriate error messages are displayed in these scenarios.
8. Repeat steps 1-4 for sign-in functionality.

#### Questions:

* Are there any specific requirements for error handling during sign-up and sign-in?

**Please review these changes and provide feedback.**

### CODE REVIEW

**Comment by Latiah** [comment]
Good work I was able to sign up and log in

**Comment by Latiah** [comment]
Hey @911samuel here are some few suggestions. Can you change so that when a person signs up ,he /she is directed to the sign up instead of going to the fill in your profile screen and also consider informing the user to enter their credentials(email or password)  when they try to sign up or in without filling them in.

**Comment by 911samuel** [comment]
> Hey @911samuel here are some few suggestions. Can you change so that when a person signs up ,he /she is directed to the sign up instead of going to the fill in your profile screen and also consider informing the user to enter their credentials(email or password) when they try to sign up or in without filling them in.

Thank you for your feedback, @Latiah. The issue was setting up Supabase to integrate it into our project. The sign-up and sign-in functionalities were simple demonstrations to show that Supabase is configured and ready to be used as our backend service. We will need someone to refine the sign-in and sign-up processes and also handle the forgot password functionality and other aspects of the authentication workflow

File: app/supabase.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { createClient } from "@supabase/supabase-js";
+
+const supabaseUrl = "https://vaxjykeihsernxqxeyiy.supabase.co";
+const supabaseAnonKey =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZheGp5a2VpaHNlcm54cXhleWl5Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTc0NDAzMTksImV4cCI6MjAzMzAxNjMxOX0.PbKiCmDG-U_1H0gkelN3_ze8_VE8lqIsc3pi5IH2bBA";
```
Comment (fredshema) [inline app/supabase.tsx]: @911samuel can you add these in environment variables, they can be shared privately but not directly in the codebase

Comment (911samuel) [reply app/supabase.tsx]: Thank you shema for the best advice and i worked on it you can now review again

**Review by Latiah — APPROVED** [review]

**Review by chrysloi — APPROVED** [review]
Looks good!

**Comment by Adadie** [comment]
Hey @911samuel, why do I get below screen when I run the app?
Screenshot_20240607_090301_Expo Go

**Comment by 911samuel** [comment]
> Hey @911samuel, why do I get below screen when I run the app? Screenshot_20240607_090301_Expo Go

All things are solved Please review again

**Pull request merged**